### ISSUE TITLE
Herbie crashing

### ISSUE THREAD

**Issue by jp2nyy** [blaylockbk/Herbie#389]
Tonight/this morning, Herbie kept on crashing on Google Colab where it says my session crashed for an unknown reason. I noticed that it took a bit longer on the herbie.toolbox import I had in my code and it stopped right after calling for a certain variable.

**Comment by fleegs79** [comment]
I've been running into the same thing with Google Colab starting on Nov 25th (worked fine on the 23rd and for the last year plus - Thank you for all of your work Brian!).  

I added some logging to the Core.py file and it's bombing during the xarray function, specifically with cfgrib.open_datasets on line 1137.  I did try manually adding that same cfgrib.open_datasets to my Colab script and was able to get it to run fine, so I'm not sure what changed in Google Colab on the 25th to cause the issue.  I did [open a ticket with Google Colab](https://github.com/googlecolab/colabtools/issues/4969), but I haven't received a response back yet.

**Comment by jp2nyy** [comment]
cfgrib gives me fits when I tried xarray originally in reading grib2 files.

**Comment by fleegs79** [comment]
I think I have it tracked down to the [ecCodes package](https://confluence.ecmwf.int/display/ECC/ecCodes+Home). The version 2.39.0 doesn't work with Herbie (at least in Google Colab), but if I role it back to 2.38.3 everything runs fine.  

For those using Google Colab, add this before installing herbie-data via pip:
!pip install eccodes==2.38.3

Hope that helps get your scripts working again. 

fleegs79

**Comment by jp2nyy** [comment]
By George, the eccodes 2.38.3 worked so far.

**Comment by blaylockbk** [comment]
Thanks for tracking this down. Should an issue be opened with cfgrib?

**Comment by blaylockbk** [comment]
yeah, this crash must just be in colab?? I can't replicate with a fresh install on linux (WSL). (I actually haven't ever used colab.)


Create new env
```bash
mamba env create -n herbie_fresh herbie-data
mamba activate herbie_fresh
```

Some versions
```
# Name                    Version                   Build    Channel
python                    3.13.1       ha99a958_102_cp313    conda-forge
herbie-data               2024.8.0           pyhd8ed1ab_0    conda-forge
eccodes                   2.39.0               hf413ef6_1    conda-forge
python-eccodes            2.37.0          py313ha014f3b_0    conda-forge
cfgrib                    0.9.14.1           pyhd8ed1ab_0    conda-forge
xarray                    2024.11.0          pyhd8ed1ab_0    conda-forge
pandas                    2.2.3           py313ha87cce1_1    conda-forge
numpy                     2.1.3           py313h4bf6692_0    conda-forge
```

Read single GRIB message in xarray
```python
from herbie import Herbie
ds = Herbie('2024-01-01').xarray('TMP:2 m')
#^ Had no issues
```

**Comment by fleegs79** [comment]
Thanks for testing that out Brian!  Maybe it's something with the older python/numpy that Google Colab uses?  Here's the same list for the scripts that crash (only eccodes gets rolled back to make it work). 

```
# Name                    Version
python                    3.10.12
herbie-data               2024.8.0 
eccodes                   2.39.0
python-eccodes            Not installed
cfgrib                    0.9.14.1
xarray                    2024.10.0 
pandas                    2.2.2
numpy                     1.26.4 
```
They are planning on updating python to 3.11 in January.

**Comment by dgketchum** [comment]
I had the same issue on my local Ubuntu machine on conda env with herbie installed today, sigkill 139 on running H.xarray, FWIW. Resolved by running 'pip install eccodes==2.38.3', as mentioned by @fleegs79.

Awesome project!
